packages/origin-ui-core/src/features/certificates/sagas.ts

```diff
@@ -288,6 +288,63 @@ function* requestPublishForSaleSaga(): SagaIterator {
     }
 }
 
+function* requestDepositSaga(): SagaIterator {
+    while (true) {
+        const action: IRequestPublishForSaleAction = yield take(
+            CertificatesActions.requestDepositCertificate
+        );
+        const exchangeClient: IExchangeClient = yield select(getExchangeClient);
+
+        if (!exchangeClient.getAllTransfers) {
+            continue;
+        }
+
+        const shouldContinue: boolean = yield call(assertCorrectBlockchainAccount);
+
+        if (!shouldContinue) {
+            continue;
+        }
+
+        const { amount, certificateId, callback, source } = action.payload;
+
+        const i18n = getI18n();
+
+        yield put(setLoading(true));
+
+        try {
+            const account: ExchangeAccount = yield call([
+                exchangeClient,
+                exchangeClient.getAccount
+            ]);
+
+            if (source === CertificateSource.Blockchain) {
```

Comment:
I guess we don’t need that check here, or if we need then move it up as a guard closure:

If (source !== ...)...

Reply:
Fixed